ec2/spark_ec2.py

```diff
@@ -983,6 +969,12 @@ def real_main():
     (opts, action, cluster_name) = parse_args()
 
     # Input parameter validation
+    if "." in opts.spark_version:
```

Comment:
Minor comment: Since we do this spark_version.replace("v", "") in two places it might make sense to keep the old function but name it as `get_spark_version` (or `check_get_spark_version`). You can then call it once here and once while constructing the template variables